torch/utils/data/distributed.py

```diff
@@ -155,25 +156,25 @@ def __init__(self, num_replicas, global_worker_rank=0, num_chunks=0, shuffle=Fal
 
     def __iter__(self):
         r"""Stride logic is implemented here"""
-        # Fecthes all indices
+        # Fetches all indices
         indices = list(self.sampler)
 
         # Create a strided sublist
-        indices = indices[self.global_worker_rank:self.num_chunks:self.num_replicas]
+        indices = indices[self.rank:self.num_chunks:self.num_replicas]
         return iter(indices)
 
     def __len__(self):
         raise NotImplementedError
 
-    def set_global_worker_rank(self, global_worker_rank):
+    def set_rank(self, rank):
         r"""Sets current global worker rank
 
         Typically used after new processes are spawned with a copy of this sampler
         to prevent sampling the same indices in different workers
         """
-        assert global_worker_rank >= 0, 'global_worker_rank must be >= 0'
-        assert global_worker_rank == 0 or global_worker_rank < self.num_replicas, 'global_worker_rank must < num_replicas'
-        self.global_worker_rank = global_worker_rank
+        assert rank >= 0, 'rank must be >= 0'
+        assert rank == 0 or rank < self.num_replicas, 'rank must < num_replicas'
```

Comment:
can this be simplified to be just 'rank < self.num_replicas'? As rank == 0 is a subset of this case?

Reply:
done